src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -2358,6 +2358,69 @@ var projectUniqueNamesForBuildIntToUpdate
                         }
                     }
 
+#if PACKAGE_CONFIG_LOCK
+                    try
+                    {
+                        if (msbuildProject != null)
+                        {
+                            //start batch processing for msbuild
+                            await msbuildProject.ProjectSystem.BeginProcessingAsync();
+                        }
+
+                        //todo: use same logic as PackageReference code to enable packages lock file (msbuild prop or file already exists)
+                        var useLockFile = false;
+                        var property = msbuildProject?.ProjectSystem?.GetPropertyValue("RestorePackagesWithLockFile");
+                        if (property is string propertyValue)
+                        {
+                            bool.TryParse(propertyValue, out useLockFile);
+                        }
+                        if (useLockFile)
+                        {
+                            //todo: get lock file name same way PackageReference projects do
+                            var directory = Path.GetDirectoryName(msbuildProject.MSBuildProjectPath);
+                            var lockFileFullPath = Path.Combine(directory, PackagesLockFileFormat.LockFileName);
+                            var lockFileName = Path.GetFileName(lockFileFullPath);
```

Comment:
you can use `PackagesLockFileUtilities.GetNuGetLockFilePath` or add an override api there for your purpose. But this will completely ignore custom lock file names.